### PR TITLE
Consolr updates

### PR DESCRIPTION
**Pull request by defect**
In order to support more tools than ipmitool in consolr, I'd like to modularize it a bit. As you can see here, I've moved all the ipmitool-related code in to a class called `Consolr::Runners::Ipmitool`. The command loads one or more runners dynamically from the config, and chooses the first one that reports it can run on a given node (meaning the runner's `can_run?` method returns true). 

An example use case could be an environment with mixed Openstack VMs and bare metal servers. An openstack runner would then handle all the VM assets and the ipmitool runner all the bare metal assets. This way, you can easily customize consolr to new tools by writing a `Consolr::Runners::Yourtool` class and then installing it in whichever way you prefer.

In order to maintain backwards compatibility I've made the ipmitool runner the default if no runners are specified in the config.

Finally, I'm adding an option (`-S`, or `--status`) which just returns the power status of the assets because it seemed like something that would be useful :)

@tumblr/collins

### CODE REVIEW

**Comment by byxorna** [comment]
LGTM

File: support/ruby/consolr/lib/consolr/runners/ipmitool.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                    end
+      end
+
+      def can_run? node
+        not (node.ipmi.address.empty? or node.ipmi.username.empty? or node.ipmi.password.empty?)
+      end
+
+			def verify node
+      	Net::Ping::External.new(node.ipmi.address).ping?
+			end
```
Comment (roymarantz) [inline support/ruby/consolr/lib/consolr/runners/ipmitool.rb]: [picky] whitespace

File: support/ruby/consolr/lib/consolr/runners/ipmitool.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      def initialize config
+        @ipmitool = if config.empty?
+                      '/usr/bin/ipmitool'
+                    else
+                      config
+                    end
+      end
+
+      def can_run? node
+        not (node.ipmi.address.empty? or node.ipmi.username.empty? or node.ipmi.password.empty?)
```
Comment (roymarantz) [inline support/ruby/consolr/lib/consolr/runners/ipmitool.rb]: do you want a rescue false incase node.ipmi (or some part) is nil?

File: support/ruby/consolr/lib/consolr/runners/runner.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+module Consolr
+  module Runners
+    class Runner
+      def can_run? node
+        raise 'can_run? is not implemented for this runner'
+      end
+
+			def verify node
+        raise 'verify is not implemented for this runner'
+			end
```
Comment (roymarantz) [inline support/ruby/consolr/lib/consolr/runners/runner.rb]: [picky] indentation

**Comment by roymarantz** [comment]
IMO there should be some way to force using a particular runner.
Do you want to add missing ipmitool functions in the PR (e.g. soft reboot vs hard reboot) or is that too much out of scope.
I like the approach though.

**Comment by gtorre** [comment]
Very nice. :+1:

**Comment by defect** [comment]
Added support for `--soft-reboot` and `--soft-off`.

**Comment by Primer42** [comment]
This LGTM
But, I think we should add a barebones example gem that would show people how to add their own, internal runners with custom logic.

**Comment by Primer42** [comment]
Can you also add unit tests before merging?

**Pull request merged**